gold/nanomips.cc

```diff
@@ -5863,6 +6064,102 @@ Target_nanomips<size, big_endian>::do_relax(
       // Change the state to EXPAND if we are done with relaxations.
       if (!again && (this->state_ == RELAX) && parameters->options().expand())
         this->state_ = EXPAND;
+      // Change the state to TRAMPOLINE.
```

Comment:
How do you tackle interactions between the expansion phase and the trampoline phase - so for example what happens when a PC-relative reference that was in-range during expansion goes out-of-range due to the introduction of stubs? This can happen for an LAPC[32] or BALC[16] referring to a different section as an example.

I'd expect these transforms to either be performed together in one pass, or in a loop with one feeding off the other, until things stabilize.

Reply:
Yes, you are right. We are doing `expand` after we are done with producing the `trampolines`.